test/email/email_header_update.c

```diff
@@ -0,0 +1,21 @@
+#define TEST_NO_MAIN
+#include "config.h"
+#include "acutest.h"
+#include "mutt/lib.h"
+#include "email/lib.h"
+
+void test_email_header_update(void)
+{
+  // struct ListNode *header_update(sturct ListNode *hdr, const struct Buffer *buf)
+  const char *existing_header = "X-Found: foo";
+  const char *new_value = "X-Found: 3.14";
+
+  struct ListNode *n = mutt_mem_calloc(1, sizeof(struct ListNode));
+  n->data = mutt_str_dup(existing_header);
+
+  {
+    struct ListNode *got = header_update(n, new_value);
+    TEST_CHECK(got == n);                          /* returns updated node */
+    TEST_CHECK(strcmp(got->data, new_value) == 0); /* node updated to new value */
+  }
+}
```

Comment:
Memory leaks

```
Direct leak of 16 byte(s) in 1 object(s) allocated from:
    #1 0x540163 in mutt_mem_calloc mutt/memory.c:55
    #2 0x46b243 in test_email_header_update test/email/email_header_update.c:13
    #3 0x40821f in test_do_run_ test/acutest.h:1023
    #4 0x4085ea in test_run_ test/acutest.h:1119
    #5 0x40a9c6 in main test/acutest.h:1721
    
Indirect leak of 14 byte(s) in 1 object(s) allocated from:
    #1 0x54bc6e in mutt_str_dup mutt/string.c:375  
    #2 0x4fe4bc in header_update email/email.c:209
    #3 0x46b28e in test_email_header_update test/email/email_header_update.c:17
    #4 0x40821f in test_do_run_ test/acutest.h:1023
    #5 0x4085ea in test_run_ test/acutest.h:1119
    #6 0x40a9c6 in main test/acutest.h:1721
```

Reply:
Silly, lazy me, all leaks should be addressed with d4ff2c6 (`EXTRA_CFLAGS="-fsanitize=address" EXTRA_LDFLAGS="-fsanitize=address" make test` passes for me with those changes)